test/evm/Precompiles.js

```diff
@@ -0,0 +1,209 @@
+        const scalar = new BN('7')
+
+        // Multiply the point by the scalar
+        const resultPoint = secondPoint.mul(scalar)
+
+        const result = await precompilesContract.callStatic.ecMul(
+            [
+                ethers.BigNumber.from(secondPoint.getX().toString()),
+                ethers.BigNumber.from(secondPoint.getY().toString())
+            ],
+            ethers.BigNumber.from(scalar.toString()),
+            ethers.BigNumber.from(prime.toString())
+        )
+
+        expect(result[0]).to.equal(resultPoint.getX())
+        expect(result[1]).to.equal(resultPoint.getY())
+
+    })
+
+    it("Should correctly compute the ecPairing check", async function () {
+        // zkSNARK verification with the pairing check. EIP-197: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-197.md
+        // Inputs are taken from circom's "Getting started" example using circom and snarkjs:  https://docs.circom.io/getting-started/installation/
+
+        const pa = ["0x10e0c597f83b5955dea39d1070b715e52c102ddb7e3a00168b44be8bf7119f55", "0x11661888377b0b03f2bcc23b8a351c8f3d23aabc8dc8df65c41e5d21c974b7c2"]
+        const pb = [["0x2cf266680cb145e28c214aa7942adb33928091b85ee6a7d0f54c94b6073b89d9", "0x0c27c439fc1fd3e02ab52f501d6667fa3d3e1187fafd5b2c0ac15f45e6fbeae9"],["0x0f2f9159625c763a41d9eda032e8333d34d8d33d241f040f56e168de8236146c", "0x18bcf1bf1212e5e13fea9a0f7a7a01a663782a018a8dfaf5d1cbed099d8f2c45"]]
+        const pc = ["0x01c57619c684da393e699fe9399a537a5d8d103b55151af2d750c67502a206ce", "0x171fe18647a62dbdb54ccb4ba1171ca0210715575ebd51e0abc5a2b8d9411f0c"]
+        const pd = ["0x0000000000000000000000000000000000000000000000000000000000000021"]
+
+        const result = await precompilesContract.ecPairing(pa, pb, pc, pd)
+        expect(result).to.be.true
+    })
+
+    it("Should return the correct Blake2 hash", async function() {
+        // data from EIP-152: https://eips.ethereum.org/EIPS/eip-152
+        const rounds = 12
+        const h = ["0x48c9bdf267e6096a3ba7ca8485ae67bb2bf894fe72f36e3cf1361d5f3af54fa5", "0xd182e6ad7f520e511f6c3e2b8c68059b6bbd41fbabd9831f79217e1319cde05b"]
+        const m = [
+            "0x6162630000000000000000000000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "0x0000000000000000000000000000000000000000000000000000000000000000"
+        ]
+        // const t = ["0x03000000", "0x00000000"]
+        const t = ["0x0300000000000000", "0x0000000000000000"];
+
+        const f = true;
+
+        const result = await precompilesContract.callStatic.blake2(rounds, h, m, t, f)
+
+        expect(result[0]).to.equal("0xba80a53f981c4d0d6a2797b69f12f6e94c212f14685ac4b74b12bb6fdbffa2d1")
+        expect(result[1]).to.equal("0x7d87c5392aab792dc252d5de4533cc9518d38aa8dbf1925ab92386edd4009923")
+    });
+    
```

Comment:
nit: I see this extra line before the final closing bracket in many places

Reply:
Updated.